Gigs/Gigs/Model Controllers/GigController.swift

```diff
@@ -0,0 +1,213 @@
+    var gigs: [Gig] = []
+    var bearer: Bearer?
+    let baseURL = URL(string: "https://lambdagigs.vapor.cloud/api")!
+
+
+
+
+
+
+    func getAllGigs(completion: @escaping (Result<[Gig], NetworkError>) -> Void ) {
+
+        guard let bearer = bearer else {
+            NSLog("No token available")
+            completion(.failure(.noBearer))
+            return
+        }
+
+        let requestURL = baseURL.appendingPathComponent("gigs")
+
+        var request = URLRequest(url: requestURL)
+
+        request.httpMethod = HTTPMethod.get.rawValue
+
+        request.setValue("Bearer \(bearer.token)", forHTTPHeaderField: "Authorization")
+
+        URLSession.shared.dataTask(with: request) { (data, response, error) in
+
+            if let response = response as? HTTPURLResponse,
+                response.statusCode == 401 {
+                completion(.failure(.badAuth))
+                return
+            }
+
+            if let error = error {
+                NSLog("Error getting animal names: \(error)")
+                completion(.failure(.apiError))
+                return
+            }
+
+            guard let data = data else {
+                completion(.failure(.noDataReturned))
+                return
+            }
+
+            let decoder = JSONDecoder()
+
+            do{
+                let gigs = try decoder.decode([Gig].self, from: data)
+                completion(.success(gigs))
+
+
+            } catch {
+                NSLog("Error decoding gigs: \(error)")
+                completion(.failure(.noDecode))
+                return
+            }
+        }.resume()
+    }
+
+    func createGig(title: String, dueDate: Date, description: String, completion: @escaping (Error?) -> Void) {
+
+        guard let bearer = bearer else {
+            completion(NSError())
+            return
+        }
+
+        let requestURL = baseURL.appendingPathComponent("gigs")
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = HTTPMethod.post.rawValue
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.setValue("Bearer \(bearer.token)", forHTTPHeaderField: "Authorization")
+        let gig = Gig(title: title, description: description, dueDate: dueDate)
+
+        do {
+            request.httpBody = try JSONEncoder().encode(gig)
+        } catch {
+            NSLog("Error encoding Gig: \(error)")
+            completion(error)
+            return
+        }
+
+        URLSession.shared.dataTask(with: request) { (_, response, error) in
+
+            if let response = response as? HTTPURLResponse,
+                response.statusCode != 200 {
+                completion(NSError())
+                return
+            }
+
+            if let error = error {
+                NSLog("Error pushing gig: \(error)")
+                completion(error)
+                return
+            }
+
+            completion(nil)
+        }.resume()
+        self.gigs.append(gig)
```

Comment:
Wrong place to put it. Anything after resume will not be part of the URLSession data task. put it on line 107 before the completion(nil)

```
self.gigs.append(gig)
            completion(nil)
        }.resume()
```